src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerExecutor.cs

```diff
@@ -792,75 +870,13 @@ private void ExecuteBlockActionsCore<TBlockStartAction, TBlockAction, TNodeActio
 
             // Include the stateful actions.
             foreach (var startAction in startActions)
-            {
```

Comment:
nit: might be nice to move the PooledDictionary creation/population into ExecuteBlockActions as it already does the freeing

Reply:
the sets are used by both callbacks :)